morph_tool/resampling.py

```diff
@@ -29,44 +28,42 @@ def _vertex_path_lengths(points):
 
 
 def _resample_from_linear_density(points, linear_density):
-    """Creates a resampling of K-2 elements on the polyline defined by
-    N `points`, omitting the first and last values which should remain
-    unchanged. The number of elements is determined by the `linear_density`.
+    """The linear density determines the number K of the new points and point
+    properties that will be created on the polyline defined by the consecutive
+    points. The functions returns the ids and fractions, where each id i and
+    fraction t corresponds to a new point via the equation:
+
+    p' = p[i] + t * (p[i + 1] - p[i])
 
-    The new sampling is represented by the line segment parametric equation:
+    which can be also used for any value defined on points (e.g. diameters):
 
     v'[k] = v[ids[k]] + fractions[k] * (v[ids[k] + 1] - v[ids[k]])
 
-    where (v'[k], v[k]) can be either `points` or any other property mapped
-    on `points`.
+    It doesn't not return the ids and fractions for first and last points in
+    the new resampling, because they should remain unchanged. Therefore, the
+    total number of ids and fractions is K-2.
 
     Args:
-        points (np.ndarray): (N, 3) Array of consecutive points definining
+        points (np.ndarray): (N, 3) Array of consecutive points defining
         segments.
         linear_density (float): Target number of points per micron
 
     Returns:
-        tuple of numpy arrays:
-            - ids (np.array): (K-2,) Array of positional ids
-            - fractions (np.array): (K-2,) Parametric fractions
+        tuple:
+            - ids (np.array): (K-2,) int array of positional ids
+            - fractions (np.array): (K-2,) float array of fractions in the
+                range [0, 1]
 
-    Notes:
-        Given consecutive points [p0, p1, ..., pN] a resampling will create
-        new points on the polyline defined by these points. This means
-        that a new point p' can be defined as a fraction t of an existing
-        segment (p[n], p[n + 1]). Therefore, using the line segment's
-        parametric eq:
+        Using the ids and fractions, the new interior points (without first
+        and last) can be created:
 
-        p' = p[n] + t * (p[n+1] - p[n])
+        new_points = points[ids] + fractions * (points[ids + 1] - points[ids])
```

Comment:
Are these new points added to the old ones? (Some docstring below indicates that we target fewer points). 

It might help code readers to say at some point that the process is (very close to) nearest neighbour(s) interpolation.

Reply:
No, it create a new resampling on the polyline formed by the old points.

I am not sure if this is nearest neighbour interpolation. I would say it's just linear interpolation.